pandas/core/arrays/_mixins.py

```diff
@@ -240,3 +241,11 @@ def _reduce(self, name: str, skipna: bool = True, **kwargs):
         else:
             msg = f"'{type(self).__name__}' does not implement reduction '{name}'"
             raise TypeError(msg)
+
+
+class StringArrayBase(ExtensionArray):
+    pass
+
+
+class StringDtypeBase(ExtensionDtype):
```

Comment:
this could alternative live in pandas\core\dtypes\base.py (might also prevent circular imports)